internal/apis/configuration/v1/types.go

```diff
@@ -57,6 +57,9 @@ type KongClusterPlugin struct {
 	// Config contains the plugin configuration.
 	Config Configuration `json:"config,omitempty"`
 
+	// ConfigFrom references a secret containing the plugin configuration.
+	ConfigFrom NamespacedSecretValueFromSource `json:"configFrom.secretKeyRef,omitempty"`
```

Comment:
What is the magical `configFrom.secretKeyRef`? I've never seen that before.
I tried to reproduce this but this doesn't work: https://play.golang.org/p/QZbfiwfgJOp

Even if it works, please instead declare a new struct for `ConfigFrom`. Having a clear hierarchy in Go types will keep this code maintainable.